parsl/tests/conftest.py

```diff
@@ -16,6 +19,25 @@
 logger = logging.getLogger('parsl')
 
 
+def dumpstacks(sig, frame):
+    s = ''
+    try:
+        thread_names = {thread.ident: thread.name for thread in threading.enumerate()}
+        tf = sys._current_frames()
+        for thread_id, frame in tf.items():
+            s += '\n\nThread: %s (%d)' % (thread_names[thread_id], thread_id)
+            s += ''.join(traceback.format_stack(frame))
+    except Exception:
+        s = traceback.format_exc()
+    with open(os.getenv('HOME') + '/parsl_stack_dump.txt', 'w') as f:
+        f.write(s)
+    print(s)
+
+
+import signal
+signal.signal(signal.SIGUSR1, dumpstacks)
```

Comment:
can this handler installation happen in something that only runs inside pytest rather than on import? I've spent way too much time tracking down "of course this will be safe to do at import" bugs especially in the test suite...

Maybe like:
```
@pytest.fixture(autouse=True, scope='session')
def load_dfk_session(request, pytestconfig):
```

Reply:
d1e075d. There's a pytest hook for session initialization, which, IMO, is clearer in intent than the fixture, but if you really want it to be a fixture, let me know or change directly.